### DISCUSSION TITLE
apis.StaticDirectoryHandler() serves some files but not all

### DISCUSSION THREAD

**Discussion by Plorenzo** [pocketbase/pocketbase#1249] [Q&A] [answered]
So I'm trying to create a new route that serves a site, basically equivalent to what `pb_public` does but using [Pocketbase as a framework](https://pocketbase.io/docs/use-as-framework/).

Everything works as expected, so when I run `go run main.go serve` and go to `localhost:8090` I can indeed see `index.html` being served. 

The strange thing is that some `js` files that that are used in the html are not being loaded, the server returns 404. The files do exist and they are in the specified path (even if I open index.html directly with the browser they load just fine ) so I'm not sure what could be happening here. 

Any help will be appreciated, if I'm doing anything wrong, please let me know. 

<img width="860" alt="Screenshot 2022-12-12 at 17 50 02" src="https://user-images.githubusercontent.com/4695692/207014852-b5044fb0-fc80-4548-a146-bb0446512061.png">

<details>
<summary>main.go:</summary>

```go
func main() {
	app := pocketbase.New()

	// Add base path tp serve ui app
	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.GET("/*", apis.StaticDirectoryHandler(ui.DistDirFS, false))
		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
} 
```
</details>

<details>
<summary>embed.go on ui/ folder</summary>

```go
//go:embed build
var webDir embed.FS

// DistDirFS contains the embedded dist directory files (without the "dist" prefix)
var DistDirFS = echo.MustSubFS(webDir, "build")
```
</details>

**Comment by ganigeorgiev** [comment, marked as answer]
`go:embed` by default doesn't load "hidden" files and those starting with underscore (you can find more info in https://pkg.go.dev/embed#hdr-Directives).

To enable the `_` prefixed files, you can change the go:embed comment to `//go:embed all:build`.

**Comment by Plorenzo** [comment]
Thanks! very much, that did the trick! and sorry for asking here, clearly not an issue with Pocketbase
